src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,260 @@
+                refresh_token = build_client_credentials_refresh_token(
+                    client, client_scope.scope, self.env_config.secret_key
+                )
+
+            elif grant_type == GranType.PASSWORD:
+                user_validation = self._validate_user_credentials(
+                    client, user_name, password
+                )
+                if user_validation:
+
+                    user = self._get_user_by_user_name(user_name)
+
+                    access_token = build_user_credential_access_token(
+                        user, client, self.env_config.secret_key
+                    )
+                    refresh_token = build_user_credential_refresh_token(
+                        user, client, self.env_config.secret_key
+                    )
+
+            persisted_refresh_token = self._add_refresh_token(
+                refresh_token, client.id, grant_type.value
+            )
+            self._add_access_token(persisted_refresh_token.id, access_token)
+
+            return {"refresh_token": refresh_token, "access_token": access_token}
+
+    def validate_token(self, token):
+        try:
+            jwt.decode(token, self.env_config.secret_key, algorithms="HS256")
+            return True
+
+        except jwt.ExpiredSignatureError:
+            return False
+        except jwt.InvalidSignatureError:
+            return False
+        except jwt.DecodeError:
+            return False
+
+    def refresh_token(self, refresh_token):
+        refresh_token_validation = self.validate_token(refresh_token)
+
+        if refresh_token_validation:
+
+            app_refresh_token = self._get_refresh_token_by_token(refresh_token)
+            client = self._get_client_by_id(app_refresh_token.app_client_id)
+
+            if app_refresh_token.grant_type == GranType.CLIENT_CREDENTIALS.value:
+
+                client_scope = self._get_scope_by_app_client_id(
+                    app_refresh_token.app_client_id
+                )
+
+                new_access_token = build_client_credentials_access_token(
+                    client, client_scope.scope, self.env_config.secret_key
+                )
+
+            else:
+                refresh_token_decoded = jwt.decode(
+                    refresh_token, self.env_config.secret_key, algorithms="HS256"
+                )
+
+                user = self._get_user_by_user_name(
+                    refresh_token_decoded.get("user_name")
+                )
+                new_access_token = build_user_credential_access_token(
+                    user, client, self.env_config.secret_key
+                )
+
+            self._delete_access_token_by_refresh_token_id(app_refresh_token.id)
+            self._add_access_token(app_refresh_token.id, new_access_token)
+
+            return {"access_token": new_access_token}
+
+        else:
+            self._delete_refresh_token(refresh_token)
+
+    def _validate_client_credentials(self, client_id, client_secret):
```

Comment:
Raise an error if there is aproblem instead. The code should not continue running.

Also remove the return.

Reply:
done.